botocore/handlers.py

```diff
@@ -275,8 +276,16 @@ def signature_overrides(service_data, service_name, session, **kwargs):
         logger.debug("Switching signature version for service %s "
                      "to version %s based on config file override.",
                      service_name, signature_version_override)
-        service_metadata = service_data['metadata']
-        service_metadata['signatureVersion'] = signature_version_override
+        return signature_version_override
+
+
+def disable_signing(**kwargs):
+    """
+    This handler disables request signing by setting the signer
+    name to an empty string, similar to how the signature
+    overrides above work.
+    """
+    return ''
```

Comment:
Yeah here is the PR: https://github.com/boto/botocore/pull/433

So in the CLI to turn off signing, you change the ``Service`` object's ``signature_version`` attribute to ``None``. So when an endpoint is created using the ``Service`` object it will pass through ``None`` to ``EndpointCreator``'s ``create_endpoint`` call to signify not to look for a signature version in the endpoint resolver or model:
https://github.com/boto/botocore/pull/433/files#diff-054181c44857a41f4e27bab9c8a6ba6eR333

But, I suppose both ``None`` and ``''`` will work for specifying no signing. Now that I think about it, the second part of my comment is a little off topic. What I really was getting at is it would be nice that we are consistent on what value disables signing ``None`` or ``''``.